snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```diff
@@ -0,0 +1,204 @@
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.snowflake.entities.SnowflakeSchema;
+import org.apache.iceberg.snowflake.entities.SnowflakeTable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SnowflakeCatalog extends BaseMetastoreCatalog
+    implements Closeable, SupportsNamespaces, Configurable<Object> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SnowflakeCatalog.class);
+
+  private Object conf;
+  private String catalogName = SnowflakeResources.DEFAULT_CATALOG_NAME;
+  private Map<String, String> catalogProperties = null;
+  private FileIO fileIO;
+  private QueryFactory queryFactory;
+
+  public SnowflakeCatalog() {}
+
+  public void setQueryFactory(QueryFactory factory) {
+    queryFactory = factory;
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    Preconditions.checkArgument(
+        namespace.length() <= SnowflakeResources.MAX_NAMESPACE_DEPTH,
+        "Snowflake doesn't supports more than 2 levels of namespace");
+
+    List<SnowflakeTable> sfTables = queryFactory.listIcebergTables(namespace);
+
+    return sfTables.stream()
+        .map(
+            table ->
+                TableIdentifier.of(table.getDatabase(), table.getSchemaName(), table.getName()))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    return false;
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {}
+
+  @Override
+  public void initialize(String name, Map<String, String> properties) {
+    catalogProperties = properties;
+
+    String uri = properties.get(CatalogProperties.URI);
+    Preconditions.checkNotNull(uri, "JDBC connection URI is required");
+
+    if (name != null) {
+      this.catalogName = name;
+    }
+
+    LOG.debug("Connecting to JDBC database {}", properties.get(CatalogProperties.URI));
+
+    JdbcClientPool connectionPool = new JdbcClientPool(uri, properties);
+
+    if (queryFactory == null) {
+      queryFactory = new SnowflakeQueryFactory(connectionPool);
+    }
+
+    String fileIOImpl = SnowflakeResources.DEFAULT_FILE_IO_IMPL;
+
+    if (null != catalogProperties.get(CatalogProperties.FILE_IO_IMPL)) {
+      fileIOImpl = catalogProperties.get(CatalogProperties.FILE_IO_IMPL);
+    }
+
+    fileIO = CatalogUtil.loadFileIO(fileIOImpl, catalogProperties, conf);
+  }
+
+  @Override
+  public void close() {
+    queryFactory.close();
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {}
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    Preconditions.checkArgument(
+        namespace.length() <= SnowflakeResources.MAX_NAMESPACE_DEPTH,
+        "Snowflake doesn't supports more than 2 levels of namespace");
+    List<Namespace> namespaceList = Lists.newArrayList();
```

Comment:
IIRC it started mostly as a pre-Java9 convention where you'd have to duplicate the generic type like "new ArrayList<Namespace>()", and also to avoid having to import both java.util.List and java.util.ArrayList; in this case I guess we can just follow whatever the rest of the Iceberg project is doing for consistency.

    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ grep -I -R Lists.newArrayList * | wc -l
    3050
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ grep -I -R "new ArrayList" * | wc -l
    22
    [dhuo@SDP_DevVM-dhuo ~/Snowflake-Labs/iceberg]$ grep -I -R "new ArrayList<>" * | wc -l
    10

Apparently they also helped make it explicit by encoding it in their checkstyle rules, so that's reassuring that we'll have some compiler guidance on project style-preferences:

https://github.com/apache/iceberg/blob/86156757653112b1a89d7e7f08903ef746fd918b/.baseline/checkstyle/checkstyle.xml#L60

    <module name="RegexpSingleline">
        <property name="format" value="new ArrayList&lt;&gt;\(.*\)"/>
        <property name="message" value="Prefer using Lists.newArrayList() instead."/>
    </module>

Reply:
Ok, thanks.